### ISSUE TITLE
OPNSense 20.1 no longer loads all the correct modules

### ISSUE THREAD

**Issue by pvols1979** [aus/pfatt#41]
FYI, OPNSense 20.1 broke this setup for me.  They are no longer loading the netgraph modules by default.  The workaround is to create a loader.conf.local file and add all of the required modules.  Here is the link to the discussion on the OPNSense forums:

https://forum.opnsense.org/index.php?topic=15653.0

**Comment by justinthegeek** [comment]
To add to what @pvols1979 said, create the file `/boot/loader.conf.local ` and add the following lines:

```
netgraph_load="YES"
ng_ether_load="YES"
ng_eiface_load="YES"
ng_one2many_load="YES"
ng_vlan_load="YES"
ng_etf_load="YES"
```
After following the rest of the guide, everything will work as expected.

**Comment by shize9ine** [comment]
Do you have to then add execute permission to the file?
`ssh root@OPNsense chmod +x /boot/loader.conf.local`

**Comment by justinthegeek** [comment]
> Do you have to then add execute permission to the file?
> `ssh root@OPNsense chmod +x /boot/loader.conf.local`

Nope!

**Comment by pvols1979** [comment]
I did not make the file executable. It was created with default permissions
on OPNsense.

**Comment by shize9ine** [comment]
Makes sense since it is not a script file. I figured configs would only need read but wasn't sure. Thanks for the quick reply.

**Comment by shize9ine** [comment]
Had issues with the /boot/loader.conf.local 
Fixed the issue for OPNsense 20.1 future users.
Modify the 99-pfatt.sh in the early folder.

`#!/bin/sh
set -e

ONT_IF='em0'
RG_IF='em1'
RG_ETHER_ADDR='10:93:97:8e:35:01'
OPNSENSE='yes'
LOG=/var/log/pfatt.log

getTimestamp(){
    echo `date "+%Y-%m-%d %H:%M:%S :: [pfatt.sh] ::"`
}

{
    echo "$(getTimestamp) pfSense + AT&T U-verse Residential Gateway for true bridge mode"
    echo "$(getTimestamp) Configuration: "
    echo "$(getTimestamp)        ONT_IF: $ONT_IF"
    echo "$(getTimestamp)         RG_IF: $RG_IF"
    echo "$(getTimestamp) RG_ETHER_ADDR: $RG_ETHER_ADDR"
    echo "$(getTimestamp)      OPNSENSE: $OPNSENSE"

    echo -n "$(getTimestamp) loading netgraph kernel modules... "
	/sbin/kldload -nq netgraph
	echo -n "$(getTimestamp) loaded netgraph"
	/sbin/kldload -nq ng_ether
	echo -n "$(getTimestamp) loaded ether"
	/sbin/kldload -nq ng_eiface
	echo -n "$(getTimestamp) loaded eiface"
	/sbin/kldload -nq ng_one2many
	echo -n "$(getTimestamp) loaded one2many"
	/sbin/kldload -nq ng_vlan
	echo -n "$(getTimestamp) loaded vlan"
	/sbin/kldload -nq ng_etf
	echo -n "$(getTimestamp) loaded etf"
    echo -n "$(getTimestamp) OK!"

    if [ ${OPNSENSE} != 'yes' ]; then
        echo -n "$(getTimestamp) attaching interfaces to ng_ether... "
        /usr/local/bin/php -r "pfSense_ngctl_attach('.', '$ONT_IF');" 
        /usr/local/bin/php -r "pfSense_ngctl_attach('.', '$RG_IF');"
        echo "OK!"
    fi 

    echo "$(getTimestamp) building netgraph nodes..."

    echo -n "$(getTimestamp)   creating ng_one2many... "
    /usr/sbin/ngctl mkpeer $ONT_IF: one2many lower one
    /usr/sbin/ngctl name $ONT_IF:lower o2m
    echo "OK!"

    echo -n "$(getTimestamp)   creating vlan node and interface... "
    /usr/sbin/ngctl mkpeer o2m: vlan many0 downstream
    /usr/sbin/ngctl name o2m:many0 vlan0
    /usr/sbin/ngctl mkpeer vlan0: eiface vlan0 ether

    /usr/sbin/ngctl msg vlan0: 'addfilter { vlan=0 hook="vlan0" }'
    /usr/sbin/ngctl msg ngeth0: set $RG_ETHER_ADDR
    echo "OK!"

    echo -n "$(getTimestamp)   defining etf for $ONT_IF (ONT)... "
    /usr/sbin/ngctl mkpeer o2m: etf many1 downstream
    /usr/sbin/ngctl name o2m:many1 waneapfilter
    /usr/sbin/ngctl connect waneapfilter: $ONT_IF: nomatch upper
    echo "OK!"

    echo -n "$(getTimestamp)   defining etf for $RG_IF (RG)... "
    /usr/sbin/ngctl mkpeer $RG_IF: etf lower downstream
    /usr/sbin/ngctl name $RG_IF:lower laneapfilter
    /usr/sbin/ngctl connect laneapfilter: $RG_IF: nomatch upper
    echo "OK!"

    echo -n "$(getTimestamp)   bridging etf for $ONT_IF <-> $RG_IF... "
    /usr/sbin/ngctl connect waneapfilter: laneapfilter: eapout eapout
    echo "OK!"

    echo -n "$(getTimestamp)   defining filters for EAP traffic... "
    /usr/sbin/ngctl msg waneapfilter: 'setfilter { matchhook="eapout" ethertype=0x888e }'
    /usr/sbin/ngctl msg laneapfilter: 'setfilter { matchhook="eapout" ethertype=0x888e }'
    echo "OK!"

    echo -n "$(getTimestamp)   enabling one2many links... "
    /usr/sbin/ngctl msg o2m: setconfig "{ xmitAlg=2 failAlg=1 enabledLinks=[ 1 1 ] }"
    echo "OK!"

    echo -n "$(getTimestamp)   removing waneapfilter:nomatch hook... "
    /usr/sbin/ngctl rmhook waneapfilter: nomatch
    echo "OK!"

    echo -n "$(getTimestamp) enabling $RG_IF interface... "
    /sbin/ifconfig $RG_IF up
    echo "OK!"

    echo -n "$(getTimestamp) enabling $ONT_IF interface... "
    /sbin/ifconfig $ONT_IF up
    echo "OK!"

    echo -n "$(getTimestamp) enabling promiscuous mode on $RG_IF... "
    /sbin/ifconfig $RG_IF promisc
    echo "OK!"

    echo -n "$(getTimestamp) enabling promiscuous mode on $ONT_IF... "
    /sbin/ifconfig $ONT_IF promisc
    echo "OK!"

    echo "$(getTimestamp) ngeth0 should now be available to configure as your pfSense WAN"
    echo "$(getTimestamp) done!"
} >> $LOG
`

**Comment by dkowis** [comment]
To chime in, I have done the `/boot/loader.conf.local` and after a reboot, I still have internet on 20.1.
